pkg/common/resources/resources.go

```diff
@@ -144,7 +144,20 @@ func (r *Resource) Clone() *Resource {
 	return ret
 }
 
-// Add additional resource to the base updating the base resource
+// Prune removes any resource type that has a zero value set.
+// Note that a zero value set and a type not being set are interpreted differently for quotas.
+func (r *Resource) Prune() {
```

Comment:
One question about the Prune, do we want to support other resources prune, for example:

When the maxapps is zero, do we want to prune it?

Reply:
I have thought about the allocated resources for the queues those could be pruned to show the type is no longer there. That would clean up the UI etc nicely. Follow up jira?

maxapps is not pruneable as the "nil" value for an integer type is 0 so we do not gain anything with that.

Reply:
Got it @wilfred-s , thanks the clarify, i create a jira to follow up:
https://issues.apache.org/jira/browse/YUNIKORN-2802